pkg/operator/ceph/cluster/mon/drain_test.go

```diff
@@ -124,41 +145,77 @@ func TestReconcileMonPDB(t *testing.T) {
 }
 
 func TestAllowMonDrain(t *testing.T) {
+	fakeNamespaceName := types.NamespacedName{Namespace: mockNamespace, Name: monPDBName}
+	// check for PDBV1 version
 	c := createFakeCluster(t, &cephv1.CephCluster{
 		Spec: cephv1.ClusterSpec{
 			DisruptionManagement: cephv1.DisruptionManagementSpec{
 				ManagePodBudgets: true,
 			},
 		},
+	}, "v1.21.0")
+	t.Run("allow mon drain", func(t *testing.T) {
+		// change MaxUnavailable mon PDB to 1
+		err := c.allowMonDrain(fakeNamespaceName)
+		assert.NoError(t, err)
+		existingPDBV1 := &policyv1.PodDisruptionBudget{}
+		err = c.context.Client.Get(context.TODO(), fakeNamespaceName, existingPDBV1)
+		assert.NoError(t, err)
+		assert.Equal(t, 1, int(existingPDBV1.Spec.MaxUnavailable.IntValue()))
 	})
-	fakeNamespaceName := types.NamespacedName{Namespace: mockNamespace, Name: monPDBName}
+	// check for PDBV1Beta1 version
+	c = createFakeCluster(t, &cephv1.CephCluster{
+		Spec: cephv1.ClusterSpec{
+			DisruptionManagement: cephv1.DisruptionManagementSpec{
+				ManagePodBudgets: true,
+			},
+		},
+	}, "v1.20.0")
 	t.Run("allow mon drain", func(t *testing.T) {
 		// change MaxUnavailable mon PDB to 1
 		err := c.allowMonDrain(fakeNamespaceName)
 		assert.NoError(t, err)
-		existingPDB := &policyv1beta1.PodDisruptionBudget{}
-		err = c.context.Client.Get(context.TODO(), fakeNamespaceName, existingPDB)
+		existingPDBV1Beta1 := &policyv1beta1.PodDisruptionBudget{}
+		err = c.context.Client.Get(context.TODO(), fakeNamespaceName, existingPDBV1Beta1)
 		assert.NoError(t, err)
-		assert.Equal(t, 1, int(existingPDB.Spec.MaxUnavailable.IntValue()))
+		assert.Equal(t, 1, int(existingPDBV1Beta1.Spec.MaxUnavailable.IntValue()))
 	})
 }
 
 func TestBlockMonDrain(t *testing.T) {
+	fakeNamespaceName := types.NamespacedName{Namespace: mockNamespace, Name: monPDBName}
+	// check for PDBV1 version
 	c := createFakeCluster(t, &cephv1.CephCluster{
 		Spec: cephv1.ClusterSpec{
 			DisruptionManagement: cephv1.DisruptionManagementSpec{
 				ManagePodBudgets: true,
 			},
 		},
+	}, "v1.21.0")
+	t.Run("block mon drain", func(t *testing.T) {
```

Comment:
Update to test name.
Suggestion:  add k8s version to both the test

```suggestion
	t.Run("block mon drain for K8s version v1.21.0", func(t *testing.T) {
```